tests/test_http_parser.py

```diff
@@ -710,6 +808,31 @@ def test_http_request_upgrade(parser: Any) -> None:
     assert tail == b"some raw data"
 
 
+def test_http_request_parser_utf8_request_line(parser: Any) -> None:
+    if not isinstance(response, HttpResponseParserPy):
```

Comment:
@pajod FYI it's much cleaner to use the `@pytest.mark.xfail` decorator since it allows pytest to make the decisions earlier in the process.

Reply:
Yeah, I noticed there was no coverage on these tests for some reason. Something to look at later.

Reply:
Yep. I started looking into the coverage drop and that's how I noticed this thing.

FWIW we _really_ should refactor how the tests with and without extensions are parametrized/generated, like I did in multidict recently. Essentially, there was an import loop in some places in tests that prevented C-extension tests from being executed (https://github.com/aio-libs/multidict/issues/837 / https://github.com/aio-libs/multidict/pull/915). So I fixed that by having an explicit global option for requiring one mode or the other, with a collection of fixtures reused everywhere and zero magic around import attempts and handling failures in weird ways.

Another thing I configured is the module classification in Codecov with different expected coverage thresholds — the goal should be that tests get 100% coverage in every CI run (from all jobs combined, of course). And the actual project code coverage should be measured as a separate metric. Currently, a global threshold value allows coverage to drop in tests if it's compensated by coverage in the project, meaning that we may be gaining more dead code (read: tests that are never executed), which results in a false sense of things being tested, when they aren't.